woocommerce/woocommerce
# Add query for product filters

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





@imanish003  @kmanijak - roping y'all in this PR because you have the most knowledge in this code space.

This PR is extraction of a larger PR here https://github.com/woocommerce/woocommerce/pull/52516

The reason for separating it is because some of the code changes touches on publicly available implementation. The goal of the larger PR is to add additional statuses to the Experimental Product Filters block.

Here are some high level notes for this PR:

- Added query clauses and logic for On Sale product statuses.
- Added query parameter `filter_status`. This is to eventually facilitate the combination of stock statuses and product statuses like On Sale. So the new Product Filters block will process `https://site.com/?filter_status=instock,onsale` etc.

We're mainly just making sure there are no regressions to the existing public product filters block.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Appearance -> Editor -> Template -> Product Catalog
2. Add the Product Filters Pattern
3. Ensure all filters are function as expected



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Add onsale and filter_status clauses and queries
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
